addons/mail/static/src/core/common/store_service.js

```diff
@@ -183,6 +183,36 @@ export class Store extends BaseStore {
         return tab === "chat" ? ["chat", "group"] : [tab];
     }
 
+    async onClickHandler(ev, thread, partnerId) {
+        const model = ev.target.dataset.oeModel;
+        const id = Number(ev.target.dataset.oeId);
+
+        let isHandled = false;
+        if (ev.target.closest(".o_channel_redirect") && model && id) {
+            ev.preventDefault();
+            const thread = this.Thread.insert({ model, id });
+            this.env.services["mail.thread"].open(thread);
+            isHandled = true;
+        } else if (ev.target.closest(".o_mail_redirect") && partnerId !== id) {
```

Comment:
For me condition on partnerId can be removed, it's fine to open chat with oneself, I think it was a mistake to prevent it.